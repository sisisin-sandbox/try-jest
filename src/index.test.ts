describe('foo', () => {
  it('works', () => {
    expect('bar').toBe('bar');
    expect({ a: 'a', b: 1, c: true, d: new Date(2020, 0, 2), e: null, f: undefined, g: 10 }).toMatchSnapshot();
  });

  it('matchers', () => {
    expect(() => {
      throw new Error('foo');
    }).toThrowError(Error('foo'));

    expect(Promise.resolve(1)).resolves.toBe(1);
    const mock = jest.fn();
    mock('mo');
    expect(mock).toBeCalledWith('mo');
  });

  describe('loads', () => {
    afterEach(() => {
      jest.resetModules();
    });
    it('add 1', () => {
      const { add } = require('./module1');

      expect(add(1)).toBe(1);
    });
    it('add 2', () => {
      const { add } = require('./module1');

      expect(add(2)).toBe(2);
    });
  });
});
