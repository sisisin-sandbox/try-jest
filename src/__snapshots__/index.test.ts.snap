// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foo works 1`] = `
Object {
  "a": "a",
  "b": 1,
  "c": true,
  "d": 2020-01-01T15:00:00.000Z,
  "e": null,
  "f": undefined,
  "g": 10,
}
`;
